app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/CustomDatasourceRepository.java

```diff
@@ -12,7 +12,7 @@
 public interface CustomDatasourceRepository extends AppsmithRepository<Datasource> {
     Flux<Datasource> findAllByOrganizationId(String organizationId, AclPermission permission);
 
-    Mono<Datasource> findByName(String name, AclPermission aclPermission);
+    Mono<Datasource> findByNameAndOrganizationId(String name, String orgId, AclPermission aclPermission);
```

Comment:
Minor, keeping the name consistent with the method's implementation.

```suggestion
    Mono<Datasource> findByNameAndOrganizationId(String name, String organizationId, AclPermission aclPermission);
```

Reply:
Updated code pushed!